dotcom-rendering/playwright/tests/parallel-1/sign-in-gate.e2e.spec.ts

```diff
@@ -0,0 +1,404 @@
+			await loadPageWithOverrides(page, labsArticle);
+
+			await scrollToGateForLazyLoading(page);
+
+			await expectToNotExist(page, SIGN_IN_GATE_MAIN_SELECTOR);
+		});
+
+		test('do NOT load the sign in gate on a device with an ios9 user agent string', async ({
+			page,
+		}) => {
+			const ios9UserAgent =
+				'Mozilla/5.0 (iPad; CPU OS 9_0 like Mac OS X) AppleWebKit/601.1.17 (KHTML, like Gecko) Version/8.0 Mobile/13A175 Safari/600.1.4';
+
+			await page.addInitScript((userAgent) => {
+				Object.defineProperty(window.navigator, 'userAgent', {
+					value: userAgent,
+				});
+			}, ios9UserAgent);
+
+			await loadPageWithOverrides(page, standardArticle);
+
+			const userAgent = await page.evaluate(
+				() => window.navigator.userAgent,
+			);
+
+			expect(userAgent).toEqual(ios9UserAgent);
+
+			await scrollToGateForLazyLoading(page);
+
+			await expectToNotExist(page, SIGN_IN_GATE_MAIN_SELECTOR);
+		});
+
+		test('remove gate when the dismiss button is clicked', async ({
+			page,
+		}) => {
+			await loadPageWithOverrides(page, standardArticle);
+
+			// set article count to be min number to view gate
+			await setArticleCount(page, 3);
+
+			await scrollToGateForLazyLoading(page);
+
+			await expectToBeVisible(page, SIGN_IN_GATE_MAIN_SELECTOR);
+
+			await page
+				.locator('[data-testid=sign-in-gate-main_dismiss]')
+				.click();
+
+			await expectToNotExist(page, SIGN_IN_GATE_MAIN_SELECTOR);
+		});
+
+		test('register CTA button should contain correct profile.theguardian.com href', async ({
+			page,
+		}) => {
+			await loadPageWithOverrides(page, standardArticle);
+
+			// set article count to be min number to view gate
+			await setArticleCount(page, 3);
+
+			await scrollToGateForLazyLoading(page);
+
+			await expectToBeVisible(page, SIGN_IN_GATE_MAIN_SELECTOR);
+
+			const registerHref = await page
+				.locator('[data-testid=sign-in-gate-main_register]')
+				.getAttribute('href');
+
+			expect(registerHref).toContain('profile.theguardian.com/register');
+		});
+
+		test('sign in link should contain correct profile.theguardian.com href', async ({
+			page,
+		}) => {
+			await loadPageWithOverrides(page, standardArticle);
+
+			// set article count to be min number to view gate
+			await setArticleCount(page, 3);
+
+			await scrollToGateForLazyLoading(page);
+
+			await expectToBeVisible(page, SIGN_IN_GATE_MAIN_SELECTOR);
+
+			const signInHref = await page
+				.locator('[data-testid=sign-in-gate-main_signin]')
+				.getAttribute('href');
+
+			expect(signInHref).toContain('profile.theguardian.com/signin');
+		});
+
+		test('show cmp ui when privacy settings link is clicked', async ({
+			context,
+			page,
+		}) => {
+			// For this test remove the gu-cmp-disabled cookie to ensure cmp is shown
+			await clearCookie(context, 'gu-cmp-disabled');
+
+			await loadPageWithOverrides(page, standardArticle);
+
+			await cmpAcceptAll(page);
+
+			// set article count to be min number to view gate
+			await setArticleCount(page, 3);
```

Comment:
Should we just visit the page thrice?

Reply:
We could it would make the tests slightly slower but it would also implicitly test the article count logic. I'm going to merge this as-is now as I can see Cypress is causing issues elsewhere and I'll circle back on this.